sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-redis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/redis/JedisRedisCheckpointStore.java

```diff
@@ -47,9 +48,22 @@ public Flux<PartitionOwnership> claimOwnership(List<PartitionOwnership> requeste
      */
     @Override
     public Flux<Checkpoint> listCheckpoints(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
-        return null;
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        try (Jedis jedis = jedisPool.getResource()) {
+            Set<String> members = jedis.smembers(prefix);
+            jedisPool.returnResource(jedis);
+            return Flux.fromStream(members.stream().map(json ->
+            {
+                try {
+                    return jacksonAdapter.deserialize(json, Checkpoint.class, SerializerEncoding.JSON);
+                }
+                catch (IOException e) {
+                    LOGGER.error("String could not be deserialized into Checkpoint object"); //Look into ideal way to log this error
+                }
+                return null; //Unsure about what to return at this point
```

Comment:
it should never reach this statement. I'm surprised it didn't give you an error

Reply:
I think it didn't give an error because I missed the throw keyword in the catch... Thanks for pointing this out!